packages/server/modules/workspaces/graph/resolvers/workspaceJoinRequests.ts

```diff
@@ -33,117 +36,160 @@ import { WorkspaceJoinRequestGraphQLReturn } from '@/modules/workspacesCore/help
+              }
+            },
+            WorkspaceJoinRequestGraphQLReturn
+          >({
+            getItems: getWorkspaceJoinRequestsFactory({ db }),
+            getTotalCount: countWorkspaceJoinRequestsFactory({ db })
+          })({
+            filter: {
+              userId: parent.id,
+              status: filter?.status ?? undefined
+            },
+            cursor: cursor ?? undefined,
+            limit
           })
         }
-      })
-      return await approveWorkspaceJoinRequest({
-        userId: args.input.userId,
-        workspaceId: args.input.workspaceId
-      })
-    },
-    deny: async (_parent, args) => {
-      const denyWorkspaceJoinRequest = commandFactory<DenyWorkspaceJoinRequest>({
-        db,
-        operationFactory: ({ db }) => {
-          const updateWorkspaceJoinRequestStatus =
-            updateWorkspaceJoinRequestStatusFactory({
-              db
-            })
-          const sendWorkspaceJoinRequestDeniedEmail =
-            sendWorkspaceJoinRequestDeniedEmailFactory({
-              renderEmail,
-              sendEmail,
-              getServerInfo: getServerInfoFactory({ db }),
-              getUserEmails: findEmailsByUserIdFactory({ db })
+      },
+      Mutation: {
+        workspaceJoinRequestMutations: () => ({})
+      },
+      WorkspaceJoinRequestMutations: {
+        approve: async (_parent, args) => {
+          const approveWorkspaceJoinRequest =
+            commandFactory<ApproveWorkspaceJoinRequest>({
+              db,
+              eventBus,
+              operationFactory: ({ db, emit }) => {
+                const updateWorkspaceJoinRequestStatus =
+                  updateWorkspaceJoinRequestStatusFactory({
+                    db
+                  })
+                const sendWorkspaceJoinRequestApprovedEmail =
+                  sendWorkspaceJoinRequestApprovedEmailFactory({
+                    renderEmail,
+                    sendEmail,
+                    getServerInfo: getServerInfoFactory({ db }),
+                    getUserEmails: findEmailsByUserIdFactory({ db })
+                  })
+                return approveWorkspaceJoinRequestFactory({
+                  updateWorkspaceJoinRequestStatus,
+                  sendWorkspaceJoinRequestApprovedEmail,
+                  getUserById: getUserFactory({ db }),
+                  getWorkspace: getWorkspaceFactory({ db }),
+                  getWorkspaceJoinRequest: getWorkspaceJoinRequestFactory({
+                    db
+                  }),
+                  upsertWorkspaceRole: upsertWorkspaceRoleFactory({ db }),
+                  emit
+                })
+              }
             })
+          return await approveWorkspaceJoinRequest({
+            userId: args.input.userId,
+            workspaceId: args.input.workspaceId
+          })
+        },
+        deny: async (_parent, args) => {
+          const denyWorkspaceJoinRequest = commandFactory<DenyWorkspaceJoinRequest>({
```

Comment:
I don't get how commandFactory works. It seems to wrap things in a transaction, but the transaction doesn't actually seem to be used in any way.

Reply:
![image](https://github.com/user-attachments/assets/ff50d3f7-8b06-4511-bfb6-23b2d55dba66)
shouldn't the trx be passed in place of db like so? cc @gjedlicska

Reply:
i'm also concerned if just wrapping all of the "command's" DB operations in a trx is something we even want to do. you might want to batch inserts/updates in a trx, while getting selects without one

Reply:
Your are seeing this as a changed line because I added the FF check which moves everything to the right, however this is not part of this PR.